### ISSUE TITLE
System.Data.SqlClient.Tests.SqlConnectionBasicTests.IntegratedAuthConnectionTest failed with SQL connection time out in CI jitstress job x86_arm_altjit_checked_windows_nt_corefx_jitstressregs4

### ISSUE THREAD

**Issue by jiangzeng** [dotnet/coreclr#20632]
**Job:**
[https://ci.dot.net/job/dotnet_coreclr/job/master/view/x86_arm_altjit/job/jitstress/job/x86_arm_altjit_checked_windows_nt_corefx_jitstressregs4/334/](https://ci.dot.net/job/dotnet_coreclr/job/master/view/x86_arm_altjit/job/jitstress/job/x86_arm_altjit_checked_windows_nt_corefx_jitstressregs4/334/)

**Affected tests:**
System.Data.SqlClient.Tests.SqlConnectionBasicTests.IntegratedAuthConnectionTest

**Log:**
```
System.Data.SqlClient.Tests.SqlConnectionBasicTests.IntegratedAuthConnectionTest [FAIL]
        System.Data.SqlClient.SqlException : Connection Timeout Expired.  The timeout period elapsed while attempting to consume the pre-login handshake acknowledgement.  This could be because the pre-login handshake failed or the server was unable to respond back in time.  The duration spent while attempting to connect to this server was - [Pre-Login] initialization=6992; handshake=113; 
        ---- System.ComponentModel.Win32Exception : The wait operation timed out.
  TDSServerEndPoint ExecuteReaderAsyncTest Is Server Socket Bound: True Testing connectivity to the endpoint created for the server.
        Stack Trace:
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\SqlClient\SqlInternalConnectionTds.cs(396,0): at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, Boolean applyTransientFaultHandling, String accessToken)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\SqlClient\SqlConnectionFactory.cs(133,0): at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\Common\src\System\Data\ProviderBase\DbConnectionFactory.cs(109,0): at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\ProviderBase\DbConnectionPool.cs(677,0): at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\ProviderBase\DbConnectionPool.cs(1622,0): at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\ProviderBase\DbConnectionPool.cs(1082,0): at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\ProviderBase\DbConnectionPool.cs(1049,0): at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\ProviderBase\DbConnectionFactory.cs(151,0): at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\Common\src\System\Data\ProviderBase\DbConnectionInternal.cs(325,0): at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\Common\src\System\Data\ProviderBase\DbConnectionClosed.cs(37,0): at System.Data.ProviderBase.DbConnectionClosed.TryOpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\SqlClient\SqlConnection.cs(1221,0): at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource`1 retry)
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\src\System\Data\SqlClient\SqlConnection.cs(757,0): at System.Data.SqlClient.SqlConnection.Open()
          D:\j\workspace\x86_arm_altji---e1c369bb\_\fx\src\System.Data.SqlClient\tests\FunctionalTests\SqlConnectionBasicTests.cs(36,0): at System.Data.SqlClient.Tests.SqlConnectionBasicTests.IntegratedAuthConnectionTest()
          ----- Inner Stack Trace -----
```

**Comment by BruceForstall** [comment]
@sandreenko I suggest you close this as "no repro" -- the current runs don't show this, and only show System.Net failures, which we should ignore.

**Comment by sandreenko** [comment]
@BruceForstall I saw it 11 days ago (https://ci.dot.net/job/dotnet_coreclr/job/master/view/x86_arm_altjit/job/jitstress/job/x86_arm_altjit_checked_windows_nt_corefx_jitstressregs4/389/testReport/System.Data.SqlClient.Tests/SqlConnectionBasicTests/IntegratedAuthConnectionTest/), so it still happens. 

I am planning to return to this issue soon.

**Comment by sandreenko** [comment]
We do not have altjit jobs anymore and that issue was probably fixed with #22746
